### PR TITLE
Implement formatting and string manipulation functions

### PR DESCRIPTION
**Pull request by F1r3D3v**
Things done in this PR:

### Implementation of mathematical functions:
* [`alkos/libc/math/math.cpp`](diffhunk://#diff-8e7e866d7c61fec482ce4274a768bdb7a321a6bea8a06da22fad778b38c3f819R1-R57): Implemented mathematical functions including `modf`, `isnan`, `isinf`, and `abs`.

### Implementation of string functions:
* [`alkos/libc/string/string.cpp`](diffhunk://#diff-e75e25fbe5e02a6c436073185b5c5d7180cf7f33eec4eedc61fc9616c6d047f7R1-R95): Implemented string functions including `strlen`, `strcpy`, `strncpy`, `strcmp`, `strncmp`, `strcat`, `strncat`, `strchr`, and `strrchr`.

### Implementation of formatted string output function
* [`alkos/libc/io/snprintf.cpp`](diffhunk://#diff-9260856c9f7ec3e5a4d683f9f1984af1ab74fd1fd19e37ef55acfbb2dffdb4e5): Implemented writing formatted data to a string, with a size limit.

### Build and script configuration updates:
* [`alkos/toolchains/x86_64-flags.cmake`](diffhunk://#diff-5f5bbeb1c23de2a86a69344ba4869c11f650978ad68ac5aa66c2ff6710bd990eL2-R3): Added `-mstackrealign` flag to C and C++ compiler flags for aligning stack to 16 bytes.
* [`scripts/install/run_alkos.bash`](diffhunk://#diff-a9f2cb53e4061982dbabb8f6fe5e15a478658094328d590893bd7341a9034537L11-R12): Updated QEMU arguments to include `-display default,show-cursor=on` for better display handling.

### CODE REVIEW

File: alkos/libc/io/snprintf.cpp
Comment (Jlisowskyy) [inline alkos/libc/io/snprintf.cpp:89]: Mixed snake_case with camelCase

**Review by Jlisowskyy — COMMENTED** [review]
### General Feedback  
The code looks good overall! I’ve added a few suggestions and marked some typos.  

### Suggestions for Future Improvements  
1. **Smaller PRs with Tests:**  
   - Moving forward, we should aim to create smaller PRs and add corresponding tests for each. For example, this PR could have been split into three distinct PRs, each focusing on a specific functionality:  
     - `math.h`  
     - `string.h`  
     - `snprintf` (potentially more than one PR, further breaking it down by features).  
   - Since `snprintf` and `asserts` are required for proper testing, this approach may only be feasible after merging foundational components like `asserts`, `snprintf`, and traces. After these are in place, manual testing will be possible, albeit in a limited form.  
   - Typically, you can drop a PR, create another branch on top of it, and continue working as usual.  

2. **Use of `static constexpr` Values:**  
   - Adding more `static constexpr` values with brief comments above them would significantly improve readability and maintainability.  

3. **Newline at End of Files:**  
   - Some files are missing a newline at the end of the file. Ensuring this is present helps maintain consistency and adheres to coding standards.  

4. **Style Suggestions (Google C++ Style Guide):**  
   - Constants should follow this naming pattern: `kConstantValue`.  
   - Header guards should include the full path in the project to ensure uniqueness.  
   - Local variables and function arguments should be named in `snake_case`.  
   - Variables should target the narrowest possible scope: [Google C++ Guide on Local Variables](https://google.github.io/styleguide/cppguide.html#Local_Variables).  
   - Use the `const` modifier wherever applicable: [Google C++ Guide on Use of Const](https://google.github.io/styleguide/cppguide.html#Use_of_const).  
   - Empty loops should include either empty curly braces `{}` or a `continue` statement for readability: [Google C++ Guide on Empty Loops](https://google.github.io/styleguide/cppguide.html#Formatting_Looping_Branching).  
   - Always use curly braces `{}` for `if` statements and loops, even when they are optional. This greatly improves readability and makes the code easier to modify later.  

### Final Thoughts  
The code is in a good state and is almost ready for approval. However, it would be great if you could address the style-related comments before merging. In the future, we should prioritize creating unit tests for functionalities that currently lack proper testing to ensure correctness. Additionally, periodic refactor tasks (e.g., monthly) could help improve the overall code quality.  

As was said a new commit is on its way, so I’ll hold off on approval for now. Great job so far—keep up the awesome work!

File: alkos/libc/string/string.cpp
Comment (F1r3D3v) [reply alkos/libc/string/string.cpp:40]: If n were provided (e.g., n=3), with str1="abc" and str2="abcX", the function would incorrectly return the difference between '\0' and 'X'.

File: alkos/libc/string/string.cpp
Comment (F1r3D3v) [reply alkos/libc/string/string.cpp:42]: Becuase the standard does not specify if char is signed or unsigned.

File: alkos/libc/string/string.cpp
Comment (F1r3D3v) [reply alkos/libc/string/string.cpp:55]: No, it's unnecessary.

**Comment by kryczkal** [comment]
We should have basic documentation on what the functions do, its inputs and outputs, especially in header files.
https://google.github.io/styleguide/cppguide.html#Function_Comments

We haven't really made that clear or stated specifically what should be the style of the comments so I propose we use doxygen

**Review by Jlisowskyy — APPROVED** [review]
LGTM

**Pull request merged**
